pkg/virt-api/api.go

```diff
@@ -203,17 +203,19 @@ func (app *virtAPIApp) composeSubresources() {
 		subws := new(restful.WebService)
 		subws.Doc(fmt.Sprintf("KubeVirt \"%s\" Subresource API.", version.Version))
 		subws.Path(rest.GroupVersionBasePath(version))
-
 		subresourceApp := rest.NewSubresourceAPIApp(app.virtCli, app.consoleServerPort)
 
-		subws.Route(subws.PUT(rest.ResourcePath(subresourcesvmGVR)+rest.SubResourcePath("restart")).
+		restartRouteBuilder := subws.PUT(rest.ResourcePath(subresourcesvmGVR)+rest.SubResourcePath("restart")).
 			To(subresourceApp.RestartVMRequestHandler).
+			Reads(v1.RestartOptions{}).
 			Param(rest.NamespaceParam(subws)).Param(rest.NameParam(subws)).
 			Operation("restart").
 			Doc("Restart a VirtualMachine object.").
 			Returns(http.StatusOK, "OK", nil).
 			Returns(http.StatusNotFound, "Not Found", nil).
-			Returns(http.StatusBadRequest, "Bad Request", nil))
+			Returns(http.StatusBadRequest, "Bad Request", nil)
+		restartRouteBuilder.ParameterNamed("body").Required(false)
```

Comment:
I wonder if that changes the value on the "body" Parameter, or just on a copy of it. Something to check.

Reply:
I think this is fine, because `ParameterNamed` returns a pointer https://godoc.org/github.com/emicklei/go-restful#RouteBuilder.ParameterNamed